compiler/rustc_lexer/src/cursor.rs

```diff
@@ -4,7 +4,7 @@ use std::str::Chars;
 ///
 /// Next characters can be peeked via `first` method,
 /// and position can be shifted forward via `bump` method.
-pub(crate) struct Cursor<'a> {
+pub struct Cursor<'a> {
```

Comment:
If we make this type public API, let's call `TokenIter: Iterator<Item=Token>`? It was internally facing "cursor over intput `char`s" before.

Reply:
Ah, I see that you've added Eof in the next commit, and yeah, that makes sense considering we've removed first_token.